test/views/DowntimeHistoryViewSpec.scala

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views
+
+import base.SpecBase
+import models._
+import models.responses.Downtime
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import play.api.test.Injecting
+import views.html.DowntimeHistoryView
+
+import java.time.LocalDateTime
+
+class DowntimeHistoryViewSpec extends SpecBase with Injecting {
+
+  "DowntimeHistoryView" - {
+
+    def view: DowntimeHistoryView = inject[DowntimeHistoryView]
+
+    lazy val document: Document = Jsoup.parse(view(downtimeHistory).body)
+    lazy val documentNoDowntime: Document = Jsoup.parse(view(Seq.empty).body)
+
+    "should have the correct breadcrumbs" in {
+      breadcrumb(0, document).text() mustBe "Home"
+      breadcrumb(0, document).attr("href") mustBe govukHomeLink
+      breadcrumb(1, document).text() mustBe "Service availability and Planned downtime"
+      breadcrumb(1, document).attr("href") mustBe homeLink
+      breadcrumb(2, document).text() mustBe "Downtime history"
+      breadcrumb(2, document).attr("href") mustBe downtimeHistoryLink
+    }
+
+    "should have the correct heading" in {
+      document.select("h1").first().text() mustBe messages("service.downtime.history.heading")
+    }
+
+    "should have a get help link" in {
+      document.body().select(".hmrc-report-technical-issue").first()
+        .attr("href") mustBe getHelpUrl
+    }
+
+    "when there are downtime history" - {
+
+      "should have the subheading" in {
+        document.select("h2").first().text() mustBe messages("service.downtime.history.h2")
+      }
+
+      "should have a table for downtime history with the correct heading" in {
+        document.select("div:nth-child(2) > div > table > thead > tr > th:nth-child(1)").get(0).text() mustBe messages("service.downtime.history.component.name")
+        document.select("div:nth-child(2) > div > table > thead > tr > th:nth-child(2)").get(0).text() mustBe messages("service.downtime.history.start")
+        document.select("div:nth-child(2) > div > table > thead > tr > th:nth-child(3)").get(0).text() mustBe messages("service.downtime.history.end")
+      }
+
+      "should have rows for GB Departures and Arrivals" in {
+        document.select("div:nth-child(2) > div > table > tbody > tr:nth-child(1) > td:nth-child(1)")
```

Comment:
Only thing I'd suggest here to try and make it less verbose, is to create a val for the main part of the html for each section to avoid repeating the css selector

val row = document.select("div:nth-child(2) > div > table > tbody > tr:nth-child(1)

row.select("td:nth-child(2)") mustbe
row.select("td:nth-child(3)") mustbe

Reply:
updated